liquibase-core/src/main/java/liquibase/command/core/AbstractRollbackCommandStep.java

```diff
@@ -0,0 +1,213 @@
+package liquibase.command.core;
+
+import liquibase.RuntimeEnvironment;
+import liquibase.Scope;
+import liquibase.change.core.RawSQLChange;
+import liquibase.changelog.*;
+import liquibase.changelog.filter.*;
+import liquibase.changelog.visitor.ChangeExecListener;
+import liquibase.changelog.visitor.ChangeSetVisitor;
+import liquibase.changelog.visitor.RollbackVisitor;
+import liquibase.command.*;
+import liquibase.database.Database;
+import liquibase.exception.DatabaseException;
+import liquibase.exception.LiquibaseException;
+import liquibase.executor.Executor;
+import liquibase.executor.ExecutorService;
+import liquibase.lockservice.LockService;
+import liquibase.logging.mdc.MdcKey;
+import liquibase.logging.mdc.MdcObject;
+import liquibase.logging.mdc.MdcValue;
+import liquibase.logging.mdc.customobjects.ChangesetsRolledback;
+import liquibase.resource.Resource;
+import liquibase.util.StreamUtil;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * RollbackCommandStep performs the rollback-to-tag logic. For backwards compatibility issues it is not called "RollbackToTag"
+ */
+public abstract class AbstractRollbackCommandStep extends AbstractCommandStep {
+
+    public static final CommandArgumentDefinition<String> ROLLBACK_SCRIPT_ARG;
+
+    static {
+        CommandBuilder builder = new CommandBuilder();
+        ROLLBACK_SCRIPT_ARG = builder.argument("rollbackScript", String.class)
+                .description("Rollback script to execute").build();
+    }
+
+    private enum RollbackMessageType {
+        WILL_ROLLBACK, ROLLED_BACK, ROLLBACK_FAILED
+    }
+
+    protected abstract String getOperationCommand();
```

Comment:
It is not clear what this method is supposed to do. Also is it necessary or could you just call the existing method which returns the command name?

Reply:
You are right, I"m duplicating stuff. Thanks